src/v3/src/util/idxUtils.ts

```diff
@@ -106,3 +106,29 @@ export const hasMinAuthenticatorOptions = (
 export const isAuthClientSet = (
   props: WidgetProps,
 ): props is RequiredKeys<WidgetProps, 'authClient'> => !!props.authClient;
+
+export const areTransactionsEqual = (
+  tx1: IdxTransaction | undefined,
+  tx2: IdxTransaction | undefined,
+): boolean => {
+  const tx1NextStepName = tx1?.nextStep?.name;
+  const tx2NextStepName = tx2?.nextStep?.name;
+  const tx1AuthKey = tx1 && getAuthenticatorKey(tx1);
+  const tx2AuthKey = tx2 && getAuthenticatorKey(tx2);
+  const tx1AuthId = tx1?.context.currentAuthenticator?.value?.id;
+  const tx2AuthId = tx2?.context.currentAuthenticator?.value?.id;
+
+  if (tx1NextStepName !== tx2NextStepName) {
+    return false;
+  }
+
+  if (tx1AuthKey !== tx2AuthKey) {
+    return false;
+  }
+
+  if (tx1AuthId !== tx2AuthId) {
+    return false;
+  }
```

Comment:
Oh I should have been more specific -- these early returns can go between the `const`s above. Will save a small amount of work too if we skip running `getAuthenticatorKey`

Reply:
I should have asked and clarified, I was thinking like, the logic before is technically the same as this since it short circuits 😅. 

This makes sense, updating now, thank you! 🙏